### PR TITLE
Add Github Actions Super-linter to SORMAS

### PR DESCRIPTION
**Pull request by jlewis91**
# Add Github Actions Super-linter to SORMAS

Fixes https://github.com/hzi-braunschweig/SORMAS-Project/issues/4568

## Feature Description

Adds automated lint checks to the SORMAS project which covers 'almost all' languages included in the SORMAS project. This is intended as 'guardrails' and additional project automation to automatically enforce basic questions of formatting. This Github action **does not** get everything right and is somewhat opinionated; that said, each language can be toggled on and off individually so that, for example, the SORMAS project can enforce its own conventions when it comes to Java or other intensively used languages. For everything else (.env, markdown, etc.), the super-linter offers 'smart defaults'.

## Problem Description

The SORMAS project, like most large software projects uses a primary language, in this case Java, and numerous secondary languages (SQL, R, bash, markdown, Javascript, Dockerfile, etc.); ensuring code quality and code consistency is a lot of work when done manually. This is where automated linters come in. Freqently, however, introducing new linters to a project is an opt-in process in which linters are individually asssessed for secondary programming languages and document formats. This eats up a lot of time
and does not contribute to significant value generation. With opt-in linting, most projects end up 'under-linted'.

## Proposed Change

Add Super-linter to the Sormas project as a basic quality gate for pull requests, to enforce consistent standards across many file types and increase PR review automation as a way of enabling broader open source participation in the project.

## Possible Alternatives

Manually add linters for each and every of the languages used in the SORMAS project. Or decide that formatting consistency is not necessary for the project.

## Additional Information

This PR is split into two commits, one for adding the github action so that files are linted 'on-change' and another which highlights a number of minor inconsistencies which could be eliminated through the introduction of this github action.

The [github/super-linter](https://github.com/github/super-linter) is developed as open source software by Github itself.

The following languages are currently deactivated (because linting currently fails), but could be reactivated one by one in a later stage:

```yaml
VALIDATE_BASH: false
VALIDATE_CSS: false
VALIDATE_HTML: false
VALIDATE_JAVA: false
VALIDATE_JAVASCRIPT_STANDARD: false
VALIDATE_JSCPD: false
VALIDATE_R: false
VALIDATE_RUST_2015: false
VALIDATE_RUST_2018: false
VALIDATE_SQL: false
VALIDATE_TYPESCRIPT_ES: false
VALIDATE_TYPESCRIPT_STANDARD: false
VALIDATE_XML: false
```

### CODE REVIEW

**Comment by JonasCir** [comment]
Hi @jlewis91 thank you for contributing! The team will take a closer look shortly :)
In the meantime, I kindly ask you to take care of some formalities laid out in [Contribution guidelines](https://github.com/hzi-braunschweig/SORMAS-Project/blob/8c7af3ec6fb1252127ad380f2e96cf0ad8f18691/CONTRIBUTING.md):

1. Open an associated issue: Copying the text from the PR description should be fine
2. amend your commit messages, they should contain the issue number of the issue you just created e.g., `[#1234] some message`

Thank you :rocket:

File: .github/workflows/linter.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+          # Full git history is needed to get a proper list of changed files within `super-linter`
+          fetch-depth: 0
+
+      ################################
+      # Run Linter against code base #
+      ################################
+      - name: Lint Code Base
+        uses: github/super-linter@v3.15.1
+        env:
+          DEFAULT_BRANCH: master
```
Comment (JonasCir) [inline .github/workflows/linter.yml:53]: Try to use `$default-branch` here and if this does not work `development`

**Comment by JonasCir** [comment]
Interesting, I was not ware of this github linter! The action itself looks good to me (module the one comment regarding default branch as we use `development`).

File: .github/workflows/linter.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+#############################
+# Start the job on all push #
+#############################
+on:
+  push:
+    branches-ignore: [master]
+    # Remove the line above to run when pushing to master
+  pull_request:
+    branches: [master]
```
Comment (JonasCir) [inline .github/workflows/linter.yml:22]: `branches: [ development ]`

**Comment by jlewis91** [comment]
@JonasCir Thanks for the prompt and friendly review! Made the fixes you suggested, let me know if anything else comes up (the fun part comes next...turning on the rest of the linters one by one ;) )

**Comment by JonasCir** [comment]
@jlewis91 thank you!
With the recent change of the default branch, we can see how it will [look](https://github.com/hzi-braunschweig/SORMAS-Project/pull/4563/checks?check_run_id=2003141510) 
> the fun part comes next...turning on the rest of the linters one by one ;)

I hear you :smile:

**Comment by jlewis91** [comment]
@JonasCir AFAIK, you have to look in the logs for the offending line, best case is to search for `[ERROR]` in the build

**Comment by JonasCir** [comment]
Ok fine with me :+1: (inline warnings might be the case if the files _included_ in the PR are badly styled (I saw that in the past somewhere))

**Review by JonasCir — CHANGES_REQUESTED** [review]
This is great, minor remarks and a merge conflict, otherwise lgtm.

File: CODE_OF_CONDUCT.md [hunk 3]
```diff
@@ -68,9 +68,9 @@ members of the project's leadership.
...
-available at https://www.contributor-covenant.org/version/1/4/code-of-conduct.html
-
-[homepage]: https://www.contributor-covenant.org
+available at <https://www.contributor-covenant.org/version/1/4/code-of-conduct.html>
 
 For answers to common questions about this code of conduct, see
-https://www.contributor-covenant.org/faq
+<https://www.contributor-covenant.org/faq>
+
+[homepage]: https://www.contributor-covenant.org
```
[the remark this answers is not in the document]
Comment (jlewis91) [reply CODE_OF_CONDUCT.md:76]: Good question, it's a markdown link reference, used in line 70

Syntax: https://gist.github.com/emedinaa/28ed71b450243aba48accd634679f805

File: SERVER_SETUP.md [hunk 4]
```diff
@@ -113,15 +150,18 @@ Setting Keycloak up as a standalone installation [Server Installation and Config
...
-To update the SORMAS Server run the following commands
-```shell script
+To update the SORMAS Server run the following commands:
+
+```bash
 ${ASADMIN} set-config-property --propertyName=payara.security.openid.clientSecret --propertyValue=${KEYCLOAK_SORMAS_UI_SECRET} --source=domain
 ${ASADMIN} set-config-property --propertyName=payara.security.openid.clientId --propertyValue=sormas-ui --source=domain
 ${ASADMIN} set-config-property --propertyName=payara.security.openid.scope --propertyValue=openid --source=domain
 ${ASADMIN} set-config-property --propertyName=payara.security.openid.providerURI --propertyValue=http://localhost:${KEYCLOAK_PORT}/keycloak/auth/realms/SORMAS --source=domain
+<!-- markdownlint-disable-next-line MD013 -->
```
[the remark this answers is not in the document]
Comment (jlewis91) [reply SERVER_SETUP.md]: Nice catch, fixed.

File: GUIDE_SORMAS2SORMAS_CERTIFICATE.md [hunk 5]
```diff
@@ -1,104 +1,104 @@
...
 7. After providing the requested data, the certificate will be imported to the truststore.
-8. The content of the ``server-access-data.csv`` provided together with the certificate will be copied to the ``server-list.csv`` file.  
+8. The content of the ``server-access-data.csv`` provided together with the certificate will be copied to the ``server-list.csv`` file.
 9. You may now delete the ``.crt`` and ``server-access-data.csv`` files.
 
 10. *Optional for test systems and other systems with self-signed ssl certificates* <br>
     You must import the SSL certificate of the other server into the ``cacerts.jks`` of your sormas domain.
     * For getting the SSL certificate you can use ``openssl`` <br>
         e.g.
         ```shell script
```
[the remark this answers is not in the document]
Comment (jlewis91) [reply GUIDE_SORMAS2SORMAS_CERTIFICATE.md:93]: Not sure what you mean here, can you clarify?

**Comment by jlewis91** [comment]
@JonasCir think this PR is ready now, just one last comment of yours which I wasn't sure how to resolve, let me know. :)

File: GUIDE_SORMAS2SORMAS_CERTIFICATE.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (JonasCir) [reply GUIDE_SORMAS2SORMAS_CERTIFICATE.md:93]: I think it should be `bash` instead of  shell script. bash is used as specifier in the other places I saw.

Comment (jlewis91) [inline GUIDE_SORMAS2SORMAS_CERTIFICATE.md:93]:
```suggestion
        ```bash
```

[the remark this answers is not in the document]
Comment (jlewis91) [reply GUIDE_SORMAS2SORMAS_CERTIFICATE.md]: Now I see it, thanks. Fixed.

**Comment by jlewis91** [comment]
@JonasCir  Good to go. :)

**Review by JonasCir — APPROVED** [review]
Thank you for your contribution! :rocket: 
lgtm

**Pull request merged**